### PR TITLE
Rename nbde_server_create_keys to nbde_server_rotate_keys

### PR DESCRIPTION
**Pull request by sergio-correia**
The behavior is the following:

    - if there are no keys, create them
    - if there are keys, do not touch them, unless nbde_server_create_keys is
      set to `yes', in which case the keys will be rotated and new keys will
      be created

### CODE REVIEW

**Comment by sergio-correia** [comment]
@richm: I have rebased this PR on top of master, but how can I be sure ` tests/test_nbde_server_create_keys.yml` is running in the CI?

**Comment by richm** [comment]
> @richm: I have rebased this PR on top of master, but how can I be sure ` tests/test_nbde_server_create_keys.yml` is running in the CI?

Looks like nbde_server is not enabled in CI. I'll have to enable it.

Also, the test file should be called `tests/tests_nbde_server_create_keys.yml` - see https://github.com/linux-system-roles/test-harness/blob/master/test/run-tests#L373

**Comment by richm** [comment]
don't worry about the el6 failures - nbde isn't supported on el6, but there isn't currently a way for a project to specify that for CI

**Comment by sergio-correia** [comment]
> > @richm: I have rebased this PR on top of master, but how can I be sure ` tests/test_nbde_server_create_keys.yml` is running in the CI?
> 
> Looks like nbde_server is not enabled in CI. I'll have to enable it.
> 
> Also, the test file should be called `tests/tests_nbde_server_create_keys.yml` - see https://github.com/linux-system-roles/test-harness/blob/master/test/run-tests#L373

Thanks, I have fixed the test name.

File: tests/tests_nbde_server_rotate_keys.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        recurse: no
+        patterns:
+          - '*.jwk'
+          - '.*.jwk'
+      check_mode: yes
+      register: existing_keys_rotate_not_set
+
+    - name: Check whether keys changed - nbde_server_rotate_keys not set
+      assert:
+        that: "{{ existing_keys.files == existing_keys_rotate_not_set.files }}"
```
Comment (richm) [inline tests/tests_nbde_server_rotate_keys.yml:44]: Is it necessary to have the test enclosed in `"{{ .... }}"`?  If it is not needed, I would prefer to drop it.
I know there are other places in the code where there are `when` and `that` tests that use `"` - don't worry about those places which are not directly related to this PR, but please change this and other `that` tests to not enclose the tests in `"{{ .... }}"` if they are not needed.

File: tests/tests_nbde_server_rotate_keys.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        recurse: no
+        patterns:
+          - '*.jwk'
+          - '.*.jwk'
+      check_mode: yes
+      register: existing_keys_rotate_set
+
+    - name: Check whether keys changed - nbde_server_rotate_keys set
+      assert:
+        that: "{{ existing_keys.files != existing_keys_rotate_set.files }}"
```
Comment (richm) [inline tests/tests_nbde_server_rotate_keys.yml:65]: here

File: tests/verify-role-results.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+- name: Gather state of required packages
+  package:
+    name: "{{ __nbde_server_package_list }}"
+    state: present
+  check_mode: true
+  register: packages_installed
+
+- name: Check whether required packages are installed
+  assert:
+    that: "{{ not packages_installed.changed }}"
```
Comment (richm) [inline tests/verify-role-results.yml:11]: here

File: tests/verify-role-results.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    name: "{{ __nbde_server_service }}"
+    enabled: true
+    state: started
+  check_mode: true
+  register: nbde_service_started
+
+- name: Check whether service is enabled and started when it should be
+  when: state is not defined or state != "stopped"
+  assert:
+    that: "{{ not nbde_service_started.changed }}"
```
Comment (richm) [inline tests/verify-role-results.yml:25]: here

File: tests/verify-role-results.yml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    name: "{{ __nbde_server_service }}"
+    enabled: true
+    state: stopped
+  check_mode: true
+  register: nbde_service_stopped
+
+- name: Check whether service is enabled and stopped when it should be
+  when: state is defined or state == "stopped"
+  assert:
+    that: "{{ not nbde_service_stopped.changed }}"
```
Comment (richm) [inline tests/verify-role-results.yml:39]: here

File: tests/verify-tang-results.yml [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  nbde_server_tang:
+    state: keys-created
+    keygen: "{{ __nbde_server_keygen }}"
+    keydir: "{{ __nbde_server_keydir }}"
+  check_mode: true
+  register: existing_keys
+
+- name: Check whether there are keys
+  assert:
+    that: "{{ not existing_keys.changed }}"
```
Comment (richm) [inline tests/verify-tang-results.yml:12]: here

File: tests/tests_nbde_server_rotate_keys.yml (hunk 1 above)
Comment (sergio-correia) [reply tests/tests_nbde_server_rotate_keys.yml:44]: I tested locally and it seems we don't need those in these cases, so I will be removing them.

File: tests/tests_default_vars.yml [hunk 7]
```diff
@@ -7,7 +7,7 @@
         - import_role:
             name: nbde_server
         - assert:
-            that: "{{ item }} is defined"
+            that: item is defined
```
Comment (richm) [inline tests/tests_default_vars.yml:10]: I think `"{{ item }} is defined"` is correct.  I tried a simple playbook like this:
```yaml
- name: set parameters for test
  hosts: localhost
  tasks:
  - set_fact:
      astring: "astringval"
  - set_fact:
      bstring: "bstringval"
  - debug:
      msg: "{{ item }} is defined"
    loop:
      - astring
      - bstring
      - cstring
  - assert:
      that: item is defined
    loop:
      - astring
      - bstring
      - cstring
```
This incorrectly reported that cstring is defined.
```
ok: [localhost] => (item=cstring) => {
    "ansible_loop_var": "item",
    "changed": false,
    "item": "cstring",
    "msg": "All assertions passed"
}
```

Comment (sergio-correia) [reply tests/tests_default_vars.yml:10]: Heh, I did this one last, and it seemed fishy, but passed the test. Let me revert this portion.
